domain/model/service/modelservice.go

```diff
@@ -157,3 +169,23 @@ func (s *ModelService) GetStatus(ctx context.Context) (model.StatusInfo, error)
 		Since:  now,
 	}, nil
 }
+
+// GetEnvironProviderVersion retrieves the version of the environment provider associated with the model.
+//
+// Errors:
+// - [modelerrors.NotFound]: Returned if the model does not exist.
+func (s *ModelService) GetEnvironProviderVersion(ctx context.Context) (int, error) {
+	model, err := s.modelSt.GetModel(ctx)
```

Comment:
This is very large in scope. While it give us the information we want to know we should look to introduce a new state function that just returns the cloud type for the model. Be explicit about what we want etc instead of bring in the world.

Reply:
Sure, added a method in state to retrieve cloud type only, thanks tom!